gradio/cli/commands/components/_docs_utils.py

```diff
@@ -0,0 +1,747 @@
+                new_args = extract_args(
+                    field_type,
+                    module_name_prefix,
+                    additional_interfaces,
+                    user_fn_refs,
+                    False,
+                    arg.__name__,
+                )
+
+                if len(new_args) > 0:
+                    arg_names.append(new_args)
+
+            if append:
+                arg_names.append(arg.__name__)
+                if arg.__name__ not in user_fn_refs:
+                    user_fn_refs.append(arg.__name__)
+        elif len(args) > 0:
+            if append:
+                arg_names.append(get_container_name(arg))
+            for inner_arg in list(args):
+                new_args = extract_args(
+                    inner_arg,
+                    module_name_prefix,
+                    additional_interfaces,
+                    user_fn_refs,
+                    True,
+                )
+
+                if len(new_args) > 0:
+                    arg_names.append(new_args)
+        else:
+            if append:
+                arg_names.append(get_param_name(arg))
+        return arg_names
+
+    module_name_prefix = module.__name__ + "."
+    additional_interfaces = {}
+    user_fn_refs = []
+
+    args = extract_args(
+        param,
+        module_name_prefix,
+        additional_interfaces,
+        user_fn_refs,
+    )
+
+    formatted_type = format_type(args)
+
+    return (formatted_type, additional_interfaces, user_fn_refs)
+
+
+def extract_docstrings(module):
+    docs = {}
+    for name, obj in inspect.getmembers(module):
+        # filter out the builtins etc
+        if name.startswith("_"):
+            continue
+        #  this could be expanded but i think is ok for now
+        if inspect.isfunction(obj) or inspect.isclass(obj):
+            docs[name] = {}
+
+            main_docstring = inspect.getdoc(obj) or ""
+            cleaned_docstring = str.join(
+                "\n",
+                [s for s in main_docstring.split("\n") if not re.match(r"^\S+:", s)],
+            )
+
+            docs[name]["description"] = cleaned_docstring
+            docs[name]["members"] = {}
+            docs["__meta__"] = {"additional_interfaces": {}}
+            for member_name, member in inspect.getmembers(obj):
+                if inspect.ismethod(member) or inspect.isfunction(member):
+                    # we are are only interested in these methods
+                    if (
+                        member_name != "__init__"
+                        and member_name != "preprocess"
+                        and member_name != "postprocess"
+                    ):
+                        continue
+
+                    docs[name]["members"][member_name] = {}
+
+                    member_docstring = inspect.getdoc(member) or ""
+                    hints = typing.get_type_hints(member)
```

Comment:
I think we need to do `member.__annotations__` here. 

`get_type_hints` is failing for me on python 3.9 because of the use of `|` for unions.

<img width="609" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/fc7c08ab-f64c-487c-b305-a4bfa4c9f7f6">

Reply:
Are these two separate issues? I'm not sure i'm understanding correctly.

If i change `hints = typing.get_type_hints(member)` to `hints = typing.get_type_hints(member.__annotations)` then it fails. Is there some specific issue with an earlier python version?

> get_type_hints is failing for me on python 3.9 because of the use of | for unions.

Is this unions in the docs code, in the component code or somewhere else?

I'll test on 3.9.